convex/searches.ts

```diff
@@ -43,87 +41,39 @@ export const search = action({
       totalTokens: inputTokens,
       embeddingMs,
     } = await fetchEmbedding(input);
-    const pineconeStart = Date.now();
-    const pinecone = await pineconeIndex();
-    const { matches } = await pinecone.query({
-      queryRequest: {
-        namespace: "chunks",
-        topK,
-        vector: embedding,
-      },
+
+    const result = await ctx.vectorSearch("chunkEmbedding", "by_embedding", {
+      vector: embedding,
+      limit: 16,
     });
-    if (!matches) throw new Error("Pinecone matches are empty");
-    const relatedChunks = matches.map(({ id, score }) => ({
-      id: id as Id<"chunks">,
-      score,
+
+    if (!result) throw new Error("Pinecone matches are empty");
+
+    const relatedChunks = result.map(({ _id, _score }) => ({
+      id: _id as Id<"chunkEmbedding">,
+      score: _score,
     }));
-    const queryMs = Date.now() - pineconeStart;
+
+    console.log({
+      inputTokens,
+      embeddingMs,
+    });
+
     if (searchId) {
-      await upsertVectors(
-        "searches",
-        [{ id: searchId, values: embedding, metadata: { input } }],
-        pinecone
-      );
-      const saveSearchMs = Date.now() - pineconeStart - queryMs;
-      console.log({
-        inputTokens,
-        embeddingMs,
-        queryMs,
-        saveSearchMs,
-      });
       await ctx.runMutation(internal.searches.patch, {
         id: searchId,
         patch: {
           relatedChunks,
           // stats
           inputTokens,
           embeddingMs,
-          queryMs,
-          saveSearchMs,
         },
       });
     }
     return relatedChunks;
   },
 });
 
-export const wordSearch = query({
-  args: { input: v.string(), count: v.number() },
-  handler: async (ctx, { input, count }) => {
-    const results = await ctx.db
-      .query("chunks")
-      .withSearchIndex("text", (q) => q.search("text", input))
-      .take(count);
-    return pruneNull(
-      await Promise.all(
-        results.map(async (chunk) => {
-          const source = await ctx.db.get(chunk.sourceId);
-          return source && { ...chunk, sourceName: source.name };
-        })
-      )
-    );
-  },
-});
```

Comment:
any reason to drop the word search for comparing search types? It's not embedding related, granted.